### PR TITLE
PR TASK 6 FIRST PART

### PR DESCRIPTION
**Pull request by ZaniInc**

### CODE REVIEW

File: .solhint.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+  "extends": "solhint:recommended",
+  "plugins": [],
+  "rules": {
+    "avoid-suicide": "error",
+    "avoid-sha3": "error",
+    "compiler-version": [
+      "error",
+      "0.8.14"
```
Comment (BhHaipls) [inline .solhint.json]: you have diff version with code, launch solhint for check

File: contracts/MyProxy.sol [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * User make call to logic contract via Proxy and work
+ * result of logic contract stored on Proxy this allows
+ * upgrade logic contract by replacing it on V2 etc.
+ *
+ * Proxy work's with logic contract by using low-level
+ * 'delegateCall' , difference between 'call' and 'delegateCall'
+ * when 'delegateCall' , calle using state from caller and change
+ * state of caller.
+ */
+contract MyProxy is TransparentUpgradeableProxy {
```
Comment (BhHaipls) [inline contracts/MyProxy.sol]: MyProxy it's not general contract, create folder Mock and add MyProxyMock to mock folder

File: contracts/MyToken.sol [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.7;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract MyToken is ERC20 {
+    uint256 public vestingSupply = 100000_000000000000000000;
+    constructor()ERC20("TOKEN","ERC") {
```
Comment (BhHaipls) [inline contracts/MyToken.sol]: code style is miss, should be format

File: contracts/VestingUpgradeableV2.sol [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+     * first part of task
+     */
+    modifier onlyOnce() {
+        require(!onlyOnceVar, "Error : can call only once time");
+        _;
+    }
+
+    /**
+     * @dev Set '_token' IERC20 to interact with thrid party token
+     *
```
Comment (BhHaipls) [inline contracts/VestingUpgradeableV2.sol]: you delete another documentation, should be return

File: test/VestingUpgradeableV1.test.js [hunk 5]
```diff
@@ -0,0 +1,411 @@
+const VestingV1 = artifacts.require("./VestingUpgradeable");
+const MyToken = artifacts.require("./MyToken");
+const MyProxy = artifacts.require("./MyProxy");
```
Comment (BhHaipls) [inline test/VestingUpgradeableV1.test.js]: Should use TransparentUpgradeableProxy instead MyProxy

**Review by BhHaipls — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Review by BhHaipls — APPROVED** [review]

**Review by BhHaipls — CHANGES_REQUESTED** [review]
should change name of tests

**Review by BhHaipls — APPROVED** [review]
approve, can be merge

**Review by BhHaipls — APPROVED** [review]
approved

**Pull request merged**